### DISCUSSION TITLE
Сбросить счётчик id

### DISCUSSION THREAD

**Discussion by Makhmut2445** [totumonline/totum-ru-issues-and-discussions#138] [Как это сделать?] [answered]
Такая ситуация, создали таблицу начали вносить туда данные, id стало уже 5 допустим..
Удаляем все 5 строк, след id будет 6. Можно сделать чтоб начинался с 1? Сбросить счётчик id.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
### Большой дисклеймер:
`id` уникальные и последовательные, но могут идти с ПРОПУСКАМИ!

Потому что `id` резервируется при старте транзакции и если она будет перезапущена, например при одновременном доступе, то `id` не откатывается, а берется следующий свободный.

### Второй большой дисклеймер:
Если вы задублируете `id`, то таблица перестанет быть работоспособной по SQL-ошибке. Сразу предупреждаю, что удаленными советами мы вам ее починить **не сможем.**

### Как сбросить последний `id`:

#### Подключиться к БД

Меняете `USER`, `PASSWORD`, `DB_NAME` на свои. Посмотреть их можно в `Conf.php` в корне установки.

```
psql --dbname=postgresql://USER:PASSWORD@localhost:5432/DB_NAME
```

##### Для Docker:

Сначала заходите в контейнер:
```
sudo docker exec -it $(sudo docker ps -f name=ttm-totum --quiet) /bin/bash
```
Из контейнера подключаетесь к БД, обратите внимание — хост для Docker другой:
```
psql --dbname=postgresql://totum:PASSWORD@ttm-postgres:5432/totum
```

#### Переходим в схему БД

Замените `SCHEMA_NAME` на имя вашей схемы (точка-запятая в конце строки обязательно!).

```
SET search_path TO SCHEMA_NAME;

# enter
```

#### Устанавливаем новый последний номер

Замените `NUMBER_HERE` и `NAME_TABLE` на ваши значения;

Нельзя установить на `0` те минимальное значение `1`. Те если вы хотите обнулить строки, то первая строка в этом случае будет c `id = 2`.

```
SELECT setval('NAME_TABLE_id_seq', NUMBER_HERE);

# enter
```

#### Выходим

```
\q

# enter
```

Если вы внутри контейнера Docker то из него тоже надо выйти набрав `exit` и нажав `enter`.
